submodules/0001-VST3-cmake_Add-RelWithDebInfo-and-MinSizeRel-configs.patch

```diff
@@ -0,0 +1,142 @@
+                 APPEND PROPERTY 
+diff --git a/modules/SMTG_AddVST3Library.cmake b/modules/SMTG_AddVST3Library.cmake
+index b22ac36..dd70c3b 100644
+--- a/modules/SMTG_AddVST3Library.cmake
++++ b/modules/SMTG_AddVST3Library.cmake
+@@ -42,7 +42,7 @@ function(smtg_run_vst_validator target)
+                 COMMAND 
+                     $<TARGET_FILE:validator> 
+                     $<$<CONFIG:Debug>:${PLUGIN_PACKAGE_PATH}>
+-                    $<$<CONFIG:Release>:${PLUGIN_PACKAGE_PATH}> 
++                    $<$<CONFIG:Release,RelWithDebInfo,MinSizeRel>:${PLUGIN_PACKAGE_PATH}>
+                     WORKING_DIRECTORY "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}"
+             )
+         endif(SMTG_WIN)
+diff --git a/modules/SMTG_ConfigureCmakeGenerator.cmake b/modules/SMTG_ConfigureCmakeGenerator.cmake
+index a990ff7..1674dff 100644
+--- a/modules/SMTG_ConfigureCmakeGenerator.cmake
++++ b/modules/SMTG_ConfigureCmakeGenerator.cmake
+@@ -18,7 +18,7 @@ endmacro(smtg_specify_output_directories)
+ #
+ # Specifies what build types (configurations) will be available.
+ macro(smtg_specify_build_types)
+-    set(CMAKE_CONFIGURATION_TYPES "Debug;Release")
++    set(CMAKE_CONFIGURATION_TYPES "Debug;Release;RelWithDebInfo;MinSizeRel")
+ 
+     # This variable is only meaningful to single-configuration generators (such as make and Ninja).
+     set(CMAKE_BUILD_TYPE "Debug" CACHE STRING "Build type")
+diff --git a/modules/SMTG_PlatformToolset.cmake b/modules/SMTG_PlatformToolset.cmake
+index 9bed3d5..a2e0fc5 100644
+--- a/modules/SMTG_PlatformToolset.cmake
++++ b/modules/SMTG_PlatformToolset.cmake
+@@ -73,12 +73,12 @@ macro(smtg_setup_platform_toolset)
+         else()
+             add_definitions(-D_UNICODE)
+             add_compile_options(/fp:fast)                   # Floating Point Model
+-            add_compile_options($<$<CONFIG:Release>:/Oi>)   # Enable Intrinsic Functions (Yes)
+-            add_compile_options($<$<CONFIG:Release>:/Ot>)   # Favor Size Or Speed (Favor fast code)
+-            add_compile_options($<$<CONFIG:Release>:/GF>)   # Enable String Pooling
+-            add_compile_options($<$<CONFIG:Release>:/EHa>)  # Enable C++ Exceptions
+-            add_compile_options($<$<CONFIG:Release>:/Oy>)   # Omit Frame Pointers
+-            #add_compile_options($<$<CONFIG:Release>:/Ox>)  # Optimization (/O2: Maximise Speed /0x: Full Optimization)
++            add_compile_options($<$<CONFIG:Release,RelWithDebInfo,MinSizeRel>:/Oi>)   # Enable Intrinsic Functions (Yes)
++            add_compile_options($<$<CONFIG:Release,RelWithDebInfo,MinSizeRel>:/Ot>)   # Favor Size Or Speed (Favor fast code)
++            add_compile_options($<$<CONFIG:Release,RelWithDebInfo,MinSizeRel>:/GF>)   # Enable String Pooling
++            add_compile_options($<$<CONFIG:Release,RelWithDebInfo,MinSizeRel>:/EHa>)  # Enable C++ Exceptions
++            add_compile_options($<$<CONFIG:Release,RelWithDebInfo,MinSizeRel>:/Oy>)   # Omit Frame Pointers
```

Comment:
Might want to keep frame pointers on RelWithDebInfo as it gives you better stack traces.

Reply:
Also the favour size or speed should be set to size for MinSizeRel